test/java/seedu/addressbook/storage/StorageFileTest.java

```diff
@@ -0,0 +1,96 @@
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void initStorageFile_nullFilePath_exceptionThrown() throws Exception {
+        thrown.expect(NullPointerException.class);
+        new StorageFile(null);
+    }
+
+    @Test
+    public void initStorageFile_noTxtExtension_exceptionThrown() throws Exception {
+        thrown.expect(IllegalValueException.class);
+        new StorageFile(TEST_DATA_FOLDER + "/InvalidFileName");
+    }
+
+    @Test
+    public void load_invalidFormat_exceptionThrown() throws Exception {
+        // The file contains valid xml data, but does not match the AddressBook class
+        StorageFile storage = getStorage("/InvalidData.txt");
+        thrown.expect(StorageOperationException.class);
+        storage.load();
+    }
+
+    @Test
+    public void load_validFormat() throws Exception {
+        AddressBook ab = getStorage("/ValidData.txt").load();
+        Person testPerson = new Person(new Name("John Doe"), 
+                                       new Phone("98765432", false),
+                                       new Email("johnd@gmail.com", false), 
+                                       new Address("John street, block 123, #01-01", false),
+                                       new UniqueTagList(Collections.emptySet()));
+        UniquePersonList list = new UniquePersonList(testPerson);
+
+        // ensure loaded AddressBook Object is properly constructed with sample data
+        assertTrue(ab.getAllPersons().equals(list));
+    }
+
+    @Test
+    public void save_nullAddressBook_exceptionThrown() throws Exception {
+        StorageFile storage = getStorage("/temp.txt");
+        thrown.expect(NullPointerException.class);
+        storage.save(null);
+    }
+
+    @Test
+    public void save_validAddressBook() throws Exception {
+        StorageFile storage = getStorage("/temp.txt");
+        AddressBook ab = new AddressBook();
+        Person testPerson = new Person(new Name("John Doe"), 
+                                       new Phone("98765432", false),
+                                       new Email("johnd@gmail.com", false), 
+                                       new Address("John street, block 123, #01-01", false),
+                                       new UniqueTagList(Collections.emptySet()));
+        ab.addPerson(testPerson);
+        storage.save(ab);
+
+        // ensure xml data for sample data is properly structured and saved
+        assertSaveSuccess("temp.txt", "ValidData.txt");
+    }
+
+    private StorageFile getStorage(String filename) throws Exception {
+        return new StorageFile(TEST_DATA_FOLDER + filename);
+    }
+
+    private void assertSaveSuccess(String file1, String file2) throws Exception {
+        String s1 = new String(Files.readAllBytes(Paths.get(TEST_DATA_FOLDER, file1)));
+        String s2 = new String(Files.readAllBytes(Paths.get(TEST_DATA_FOLDER, file2)));
+        assertEquals(s1, s2);
```

Comment:
If you can't find a way around this problem, I don't mind if you use the load method to verify the save method. Although not very desirable (a bug in load method can break tests on save method), it gets the job done and even can uncover unexpected bugs in the load method.
@pyokagan any input?